numpy/random/mtrand/mtrand.pyx

```diff
@@ -4836,12 +4836,13 @@ cdef class RandomState:
                     self._shuffle_raw(n, sizeof(npy_intp), stride, x_ptr, buf_ptr)
                 else:
                     self._shuffle_raw(n, itemsize, stride, x_ptr, buf_ptr)
-        elif isinstance(x, np.ndarray) and x.ndim > 1 and x.size:
-            # Multidimensional ndarrays require a bounce buffer.
-            buf = np.empty_like(x[0])
```

Comment:
I think changing this to `x[0,...]` would do the trick

Reply:
Done.